src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -597,16 +597,26 @@ public class ClusterTransactionMergedCommand : TransactionMergedCommand
             private readonly List<ClusterTransactionCommand.SingleClusterDatabaseCommand> _batch;
             public readonly Dictionary<long, DynamicJsonArray> Replies = new Dictionary<long, DynamicJsonArray>();
             public readonly Dictionary<long, ClusterTransactionCommand.ClusterTransactionOptions> Options = new Dictionary<long, ClusterTransactionCommand.ClusterTransactionOptions>();
+            public readonly string DatabaseRecordId;
+            public readonly string ClusterTransactionId;
 
             public ClusterTransactionMergedCommand(DocumentDatabase database, List<ClusterTransactionCommand.SingleClusterDatabaseCommand> batch) : base(database)
             {
                 _batch = batch;
+
+                var topology = database.ServerStore.LoadDatabaseTopology(database.Name);
```

Comment:
why do we need to do that for _every_ cluster tx command?

Reply:
done